source/module_base/math_chebyshev.cpp

```diff
@@ -49,35 +50,55 @@ void FFTW<float>::execute_fftw()
 // A number to control the number of grids in C_n integration
 #define EXTEND 16
 
-template <typename REAL>
-Chebyshev<REAL>::Chebyshev(const int norder_in) : fftw(2 * EXTEND * norder_in)
+template <typename REAL, typename Device>
+Chebyshev<REAL, Device>::Chebyshev(const int norder_in) : fftw(2 * EXTEND * norder_in)
 {
     this->norder = norder_in;
     norder2 = 2 * norder * EXTEND;
     if (this->norder < 1)
     {
-        ModuleBase::WARNING_QUIT("Stochastic_Chebychev", "The Chebyshev expansion order should be at least 1!");
+        ModuleBase::WARNING_QUIT("Chebyshev", "The Chebyshev expansion order should be at least 1!");
+    }
+    coefr_cpu = new REAL[norder];
```

Comment:
where to delete?

Reply:
It seems dangerous to write in this way if someone fails to delete it

Reply:
"new" is in constructor function and "delete" is in destructor function. It will allocate and deallocate with the creation and destruction of the class.

Reply:
> It seems dangerous to write in this way if someone fails to delete it

"new" is in constructor function and "delete" is in destructor function. It will allocate and deallocate with the creation and destruction of the class.